packages/cf.js/src/structs.ts

```diff
@@ -0,0 +1,38 @@
+import { ABIEncoding, Address } from "./simple-types";
+
+export enum NodeMessageType {
+  INSTALL = "install",
+  QUERY = "query",
+  ERROR = "error"
+}
+
+export enum QueryType {
+  GET_APP_INSTANCES = "getAppInstances"
+}
+
+export interface MessageDataQuery {
+  queryType: QueryType;
+  appInstances?: any[];
+}
+
+export interface MessageDataError {
+  message: string;
+  extra?: any;
+}
+
+export interface NodeMessage {
+  requestId: string;
+  messageType: NodeMessageType;
+  data: MessageDataQuery | MessageDataError | null;
```

Comment:
@mvanderh Instead of accepting `null` as a type, wouldn't it better to tag the property as `data?`?

Reply:
I prefer forcing the node provider to explicitly specify null if there's no data.